README
NAME

    Dist::Zilla::PluginBundle::Author::CHIM - Dist::Zilla configuration the
    way CHIM does it

VERSION

    version 0.052004

DESCRIPTION

    This is a Dist::Zilla PluginBundle. It is roughly equivalent to the
    following dist.ini:

        [Git::NextVersion]
        version_regexp = ^([\d._]+)(-TRIAL)?$
    
        [GatherDir]
        [PruneCruft]
    
        ;; modified files
        [OurPkgVersion]
    
        [PodWeaver]
        config_plugin = @CHIM
    
        [NextRelease]
        time_zone = UTC
        format    = %-7v %{EEE MMM d HH:mm:ss yyyy ZZZ}d
        [Authority]
        authority      = %{authority}
        do_metadata    = 1
        locate_comment = 1
    
        ;; generated files
        [License]
    
        [ReadmeAnyFromPod / ReadmeInBuild]
        type     = text
        filename = README
        location = build
    
        [ReadmeAnyFromPod / ReadmeMdInRoot]
        type     = markdown
        filename = README.md
        location = root
    
        [TravisCI::StatusBadge]
        vector = 1
    
        [MetaNoIndex]
        directory = t
        directory = xt
        directory = eg
        directory = examples
        directory = corpus
        package   = DB
        namespace = t::lib
    
        [GithubMeta]
        homepage = https://metacpan.org/release/%{dist}
        remote = origin
        remote = github
        remote = gh
        issues = 1
    
        ;; add 'provides' to META
        [MetaProvides::Package]
        meta_noindex = 1
    
        ;; META files
        [MetaYAML]
        [MetaJSON]
    
        ;; t tests
        [Test::Compile]
        fake_home = 1
    
        ;; xt tests
        [MetaTests]
        [PodSyntaxTests]
        [PodCoverageTests]
        [Test::Version]
        [Test::Kwalitee]
        [Test::EOL]
        [Test::NoTabs]
    
        ;; build
        [MakeMaker]
        [Manifest]
    
        ;; run tests at xt/ on dzil test
        [RunExtraTests]
        default_jobs = 7
    
        [Git::Check]
        allow_dirty = dist.ini
        allow_dirty = Changes
        untracked_files = die
    
        ;; release
        [ConfirmRelease]
        [UploadToCPAN]
    
        [Git::Commit]
        commit_msg = bump Changes v%v%t [ci skip]
    
        [Git::Tag]
        tag_format = %v%t
        tag_message = release v%v%t

SYNOPSYS

        # in dist.ini
        [@Author::CHIM]
        dist            = My-Very-Cool-Module
        authority       = cpan:CHIM
        github.user     = Wu-Wu

OPTIONS

 -remove

    Removes a plugin. Might be used multiple times.

        [@Author::CHIM]
        -remove = PodCoverageTests
        -remove = Test::Kwalitee

 dist

    The name of the distribution. Required.

 no_git

    Boolean. When true - all git-related plugins will be skipped. Default
    value is false.

 authority

    This one is used to set name the CPAN author of the distibution. It
    should be something like cpan:PAUSEID. Default value is cpan:CHIM.

 github.user

    Indicates github.com's account name. Default value is Wu-Wu. Used by
    Dist::Zilla::Plugin::GithubMeta and
    Dist::Zilla::Plugin::TravisCI::StatusBadge.

 github.repo

    Indicates github.com's repository name. Default value is set to value
    of the "dist" option. Used by Dist::Zilla::Plugin::GithubMeta and
    Dist::Zilla::Plugin::TravisCI::StatusBadge.

 NextRelease.time_zone

    Timezone for entries in Changes file. Default value is UTC.

    See more at Dist::Zilla::Plugin::NextRelease.

 NextRelease.format

    Format of entry in Changes file. Default value is %-7v %{EEE MMM d
    HH:mm:ss yyyy ZZZ}d.

    See more at Dist::Zilla::Plugin::NextRelease.

 MetaNoIndex.directory

    Exclude directories (recursively with files) from indexing by
    PAUSE/CPAN. Default values: t, xt, eg, examples, corpus. Allowed
    multiple values, e.g.

        MetaNoIndex.directory = foo/bar
        MetaNoIndex.directory = quux/bar/foo

    See more at Dist::Zilla::Plugin::MetaNoIndex.

 MetaNoIndex.namespace

    Exclude stuff under the namespace from indexing by PAUSE/CPAN. Default
    values: t::lib. Allowed multiple values, e.g.

        MetaNoIndex.namespace = Foo::Bar
        MetaNoIndex.namespace = Quux::Foo

    See more at Dist::Zilla::Plugin::MetaNoIndex.

 MetaNoIndex.package

    Exclude the package name from indexing by PAUSE/CPAN. Default values:
    DB. Allowed multiple values, e.g.

        MetaNoIndex.package = Foo::Bar

    See more at Dist::Zilla::Plugin::MetaNoIndex.

 MetaNoIndex.file

    Exclude specific filename from indexing by PAUSE/CPAN. No defaults.
    Allowed multiple values, e.g.

        MetaNoIndex.file = lib/Foo/Bar.pm

    See more at Dist::Zilla::Plugin::MetaNoIndex.

 GatherDir.exclude_match

    Regular expression pattern which causes not to gather matched files. No
    defaults. Allowed multiple values, e.g.

        GatherDir.exclude_match = ^foo.*
        GatherDir.exclude_match = ^ba(r|z)\/qux.*

    See more at Dist::Zilla::Plugin::GatherDir.

 GitNextVersion.version_regexp

    Regular expression that matches a tag containing a version. Default
    value is ^([\d._]+)(-TRIAL)?$.

    See more at Dist::Zilla::Plugin::Git::NextVersion.

 GitTag.tag_format

    Format of the tag to apply. Default value is %v%t.

    See more at Dist::Zilla::Plugin::Git::Tag.

 GitTag.tag_message

    Format of the tag annotation. Default value is release v%v%t.

    See more at Dist::Zilla::Plugin::Git::Tag.

 GitCommit.commit_msg

    The commit message to use in commit after release. Default value is
    bump Changes v%v%t [ci skip].

    See more at Dist::Zilla::Plugin::Git::Commit.

 GitCheck.allow_dirty

    File that is allowed to have local modifications. This option may
    appear multiple times. The default list is dist.ini and Changes.

    See more at Dist::Zilla::Plugin::Git::Check.

 GitCheck.untracked_files

    The commit message to use in commit after release. Default value is
    die.

    See more at Dist::Zilla::Plugin::Git::Check.

 GithubMeta.homepage

    Homepage of the distribution. Default value is
    https://metacpan.org/release/%{dist}.

    See more at Dist::Zilla::Plugin::GithubMeta.

 GithubMeta.remote

    Remote names to inspect for github repository. Default values are
    origin, github, gh. You can provide multiple remote names

        [@Author::CHIM]
        GithubMeta.remote = foo
        GithubMeta.remote = bar

    See more at Dist::Zilla::Plugin::GithubMeta.

 GithubMeta.issues

    Inserts a bugtracker url to metadata. Default value is 1.

    See more at Dist::Zilla::Plugin::GithubMeta.

 PodWeaver.config_plugin

    Configuration of Pod::Weaver. This option may appear multiple times.
    Default value is @CHIM.

    See more at Dist::Zilla::Plugin::PodWeaver and
    Pod::Weaver::PluginBundle::CHIM.

METHODS

 configure

    Bundle's configuration for role Dist::Zilla::Role::PluginBundle::Easy.

FAKE RELEASE

    Use option fake_release in bundle configuration:

        [@Author::CHIM]
        ...
        fake_release = 1

    or environment variable FAKE:

        FAKE=1 dzil release

    The distribution won't actually uploaded to the CPAN if option or
    variable will found.

SEE ALSO

    Please see those modules/websites for more information related to this
    module.

      * Dist::Zilla

      * Dist::Zilla::Role::PluginBundle::Easy

      * Dist::Zilla::Plugin::Authority

      * Dist::Zilla::Plugin::MetaNoIndex

      * Dist::Zilla::Plugin::NextRelease

      * Dist::Zilla::Plugin::GatherDir

      * Dist::Zilla::Plugin::Git

      * Dist::Zilla::Plugin::TravisCI::StatusBadge

      * Dist::Zilla::Plugin::GithubMeta

BUGS

    Please report any bugs or feature requests on the bugtracker website
    https://github.com/Wu-Wu/Dist-Zilla-PluginBundle-Author-CHIM/issues

    When submitting a bug or request, please include a test-file or a patch
    to an existing test-file that illustrates the bug or desired feature.

AUTHOR

    Anton Gerasimov <chim@cpan.org>

COPYRIGHT AND LICENSE

    This software is copyright (c) 2013 by Anton Gerasimov.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Perl extension Dist::Zilla::PluginBundle::Author::CHIM

0.052004 Sun Jan 4 13:21:26 2015 +0000

    [ FEATURES ]
    - added new option "no_git"

    [ FIXES ]
    - support for git-less environment (e.g. cpantesters)

0.052003 Sat Jan 3 18:41:45 2015 +0000

    [ FEATURES ]
    - removed MintingProfile::CHIM
    - extra tests located at xt/ dir now
    - added plugin to run extra tests
    - start using Bootstrap::lib

    [ TESTS ]
    - added basic tests

    [ FIXES ]
    - fixed github's repo/user parameters

0.052002 Sat Jan 3 12:25:53 2015 +0000

    [ FEATURES ]
    - PodWeaver default config added

0.052001 Sat Dec 27 17:40:03 2014 +0000

    [ FEATURES ]
    - update TravisCI::StatusBadge to 0.005
    - replace ReadmeFromPod with ReadmeAnyFromPod

0.051003 Sun Jul 20 18:51:30 2014 +0000

    [ FEATURES ]
    - added remove plugins functionality (via PluginRemover)
    - MetaResources replaced with GithubMeta
    - faking release with environment variable FAKE

0.051002 Fri Jul 18 06:33:43 2014 +0000

    [ FIXES ]
    - tagging release after changes have been commited

0.051001 Fri Jul 18 06:22:27 2014 +0000

    [ FEATURES ]
    - added plugins Git::NextVersion, Git::Tag, Git::Commit, Git::Check

0.050002 Mon Jul 14 16:17:36 2014 +0000

    [ FEATURES ]
    - added tuning for NextRelease
    - added tuning for GatherDir

    [ FIXES ]
    - update scheme in github repo link (git to https)

0.050001 Mon Jul 14 07:15:13 2014 +0000

    [ FEATURES ]
    - use v5
    - added TravisCI::StatusBadge plugin
    - added tuning options for MetaNoIndex
    - switch version to x.yyyyyy

0.04    Sun Oct 6 13:20:29 2013 +0000

    [ FEATURES ]
    - added plugin FakeRelease
    - more files to exclude via .gitignore
    - added a few versions of the perl to the .tavis.yml
    - in root directory README.pod replaced by README.md
    - added plugins Test::EOL and Test::NoTabs

    [ META ]
    - minimum perl version set to 5.8.9

0.03    Mon Mar 25 10:10:54 2013 +0000

    [ FEATURES ]
    - added basic minting profile

0.02    Wed Mar 20 15:56:30 2013 +0000

    [ META ]
    - updated prerequisites

    [ FEATURES ]
    - added plugins NextRelease, ConfirmRelease, UploadToCPAN
    - added .travis.yml

0.01    Sun Mar 17 08:22:52 2013 +0000

    - initial release


lib/Pod/Weaver/PluginBundle/CHIM.pm
package Pod::Weaver::PluginBundle::CHIM;

# ABSTRACT: CHIM's default Pod::Weaver configuration

use strict;
use warnings;

our $VERSION = '0.052004'; # VERSION
our $AUTHORITY = 'cpan:CHIM'; # AUTHORITY

use namespace::autoclean;

use Pod::Weaver 4;
use Pod::Weaver::Config::Assembler;
use Pod::Weaver::Section::SeeAlso;



sub _exp {
    my ( $package ) = @_;
    return Pod::Weaver::Config::Assembler->expand_package( $package );
}


sub mvp_bundle_config {
    return (
        [ '@CHIM/SingleEncoding',   _exp('-SingleEncoding'), {} ],
        [ '@CHIM/CorePrep',         _exp('@CorePrep'), {} ],
        [ '@CHIM/Name',             _exp('Name'),      {} ],
        [ '@CHIM/Version',          _exp('Version'),   {} ],

        [ '@CHIM/prelude',          _exp('Region'),  { region_name => 'prelude' } ],

        [ 'SYNOPSIS',               _exp('Generic'), {} ],
        [ 'DESCRIPTION',            _exp('Generic'), {} ],
        [ 'OVERVIEW',               _exp('Generic'), {} ],

        [ 'TYPES',                  _exp('Collect'), { command => 'type' } ],
        [ 'ATTRIBUTES',             _exp('Collect'), { command => 'attr' } ],
        [ 'METHODS',                _exp('Collect'), { command => 'method' } ],
        [ 'FUNCTIONS',              _exp('Collect'), { command => 'func' } ],

        [ '@CHIM/Leftovers',        _exp('Leftovers'), {} ],

        [ '@CHIM/postlude',         _exp('Region'), { region_name => 'postlude' } ],

        [ '@CHIM/SeeAlso',          _exp('SeeAlso'), {} ],
        [ '@CHIM/Bugs',             _exp('Bugs'),    {} ],
        [ '@CHIM/Authors',          _exp('Authors'), {} ],
        [ '@CHIM/Legal',            _exp('Legal'),   {} ],
    );
}


1;

__END__

=pod

=encoding UTF-8

=head1 NAME

Pod::Weaver::PluginBundle::CHIM - CHIM's default Pod::Weaver configuration

=head1 VERSION

version 0.052004

=head1 SYNOPSIS

In C<weaver.ini>

    [@CHIM]

or in C<dist.ini>

    [PodWeaver]
    config_plugin = @CHIM

=head1 DESCRIPTION

This is config for L<Pod::Weaver> I use to build documentation for my modules.

=head1 OVERVIEW

This plugin bundle is equivalent to the following C<weaver.ini> config:

    [-SingleEnconding]

    [@CorePrep]

    [Name]
    [Version]

    [Region / prelude]

    [Generic / SYNOPSIS]
    [Generic / DESCRIPTION]
    [Generic / OVERVIEW]

    [Collect / TYPES]
    command = type

    [Collect / ATTRIBUTES]
    command = attr

    [Collect / METHODS]
    command = method

    [Collect / FUNCTIONS]
    command = func

    [Leftovers]

    [Region / postlude]

    [SeeAlso]
    [Bugs]
    [Author]
    [Legal]

=for Pod::Coverage _exp

=for Pod::Coverage mvp_bundle_config

=head1 SEE ALSO

Please see those modules/websites for more information related to this module.

=over 4

=item *

L<Dist::Zilla::PluginBundle::Author::CHIM|Dist::Zilla::PluginBundle::Author::CHIM>

=item *

L<Pod::Weaver|Pod::Weaver>

=item *

L<Dist::Zilla::Plugin::PodWeaver|Dist::Zilla::Plugin::PodWeaver>

=back

=head1 BUGS

Please report any bugs or feature requests on the bugtracker website
https://github.com/Wu-Wu/Dist-Zilla-PluginBundle-Author-CHIM/issues

When submitting a bug or request, please include a test-file or a
patch to an existing test-file that illustrates the bug or desired
feature.

=head1 AUTHOR

Anton Gerasimov <chim@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2013 by Anton Gerasimov.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.029.
use strict;
use warnings;

use 5.008009;

use ExtUtils::MakeMaker;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Dist::Zilla configuration the way CHIM does it",
  "AUTHOR" => "Anton Gerasimov <chim\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-Author-CHIM",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.008009",
  "NAME" => "Dist::Zilla::PluginBundle::Author::CHIM",
  "PREREQ_PM" => {
    "Dist::Zilla" => "5.010",
    "Dist::Zilla::Plugin::Authority" => "1.006",
    "Dist::Zilla::Plugin::Bootstrap::lib" => "1.001000",
    "Dist::Zilla::Plugin::CheckExtraTests" => "0.025",
    "Dist::Zilla::Plugin::Git" => "2.023",
    "Dist::Zilla::Plugin::GithubMeta" => "0.46",
    "Dist::Zilla::Plugin::MetaProvides::Package" => "1.14000001",
    "Dist::Zilla::Plugin::OurPkgVersion" => "0.004000",
    "Dist::Zilla::Plugin::PodWeaver" => "3.101641",
    "Dist::Zilla::Plugin::ReadmeAnyFromPod" => "0.133290",
    "Dist::Zilla::Plugin::Test::Compile" => "2.001",
    "Dist::Zilla::Plugin::Test::EOL" => "0.08",
    "Dist::Zilla::Plugin::Test::Kwalitee" => "2.03",
    "Dist::Zilla::Plugin::Test::NoTabs" => "0.04",
    "Dist::Zilla::Plugin::Test::Version" => "0.002004",
    "Dist::Zilla::Plugin::TravisCI::StatusBadge" => "0.005",
    "Dist::Zilla::Role::PluginBundle::PluginRemover" => "0.102",
    "Moose" => "2.0",
    "Pod::Coverage::TrustPod" => 0,
    "Pod::Weaver::Section::SeeAlso" => "1.003",
    "Test::CPAN::Meta" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "File::Spec" => 0,
    "File::Temp" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => 0
  },
  "VERSION" => "0.052004",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla" => "5.010",
  "Dist::Zilla::Plugin::Authority" => "1.006",
  "Dist::Zilla::Plugin::Bootstrap::lib" => "1.001000",
  "Dist::Zilla::Plugin::CheckExtraTests" => "0.025",
  "Dist::Zilla::Plugin::Git" => "2.023",
  "Dist::Zilla::Plugin::GithubMeta" => "0.46",
  "Dist::Zilla::Plugin::MetaProvides::Package" => "1.14000001",
  "Dist::Zilla::Plugin::OurPkgVersion" => "0.004000",
  "Dist::Zilla::Plugin::PodWeaver" => "3.101641",
  "Dist::Zilla::Plugin::ReadmeAnyFromPod" => "0.133290",
  "Dist::Zilla::Plugin::Test::Compile" => "2.001",
  "Dist::Zilla::Plugin::Test::EOL" => "0.08",
  "Dist::Zilla::Plugin::Test::Kwalitee" => "2.03",
  "Dist::Zilla::Plugin::Test::NoTabs" => "0.04",
  "Dist::Zilla::Plugin::Test::Version" => "0.002004",
  "Dist::Zilla::Plugin::TravisCI::StatusBadge" => "0.005",
  "Dist::Zilla::Role::PluginBundle::PluginRemover" => "0.102",
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::Temp" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Moose" => "2.0",
  "Pod::Coverage::TrustPod" => 0,
  "Pod::Weaver::Section::SeeAlso" => "1.003",
  "Test::CPAN::Meta" => 0,
  "Test::More" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



